src/pages/Home.jsx

```diff
@@ -57,35 +59,55 @@ const ModalContent = styled.div`
   }
 `;
 
-const Listing = styled.div`
-  display: flex;
-  align-items: center;
-  background: white;
-  padding: 5px 10px;
-  margin: 5px 0;
-  border-radius: 8px;
-`;
-
 const Home = () => {
-  const history = useHistory();
-  const [listings, setListings] = useState([]);
+  const history = useHistory()
+  const [listings, setListings] = useState([])
   const [filters, setFilters] = useState({
     cardType: '', // 'Shop' or 'Booster'
     rarity: '', // 'epic', 'rare', 'uncommon', 'common'
     token: '', // the token's contract address
-    keyword: '' // search keyword
+    keyword: '', // search keyword,
+    sortField: '' //sort by key
   });
   const [sortBy, setSortBy] = useState({
     field: '', //attribute name of an auction
     order: 1 // 1 : ascending, -1 : descending
   })
-  const [loading, setLoading] = useState(false);
-
+  const [loading, setLoading] = useState(false)
+  const [totalCount, setTotalCount] = useState(0)
 
   const {
     state: { contracts },
   } = useContext(store);
 
+  const {
+    data,
+    isLoading,
+    isError,
+    hasNextPage,
+    fetchNextPage,
+    remove
+  } = useInfiniteQuery(
```

Comment:
Can you move this to `useListing` hook? Also I think we can directly hit the api to get open listings now right?

Reply:
ok, i will fi xit.

Reply:
I've fixed this.